addons/mrp/models/mrp_production.py

```diff
@@ -479,7 +479,22 @@ def _compute_state(self):
             elif any(not float_is_zero(move.quantity_done, precision_rounding=move.product_uom.rounding or move.product_id.uom_id.rounding) for move in production.move_raw_ids):
                 production.state = 'progress'
 
-            production._set_reservation_state()
+    @api.depends('state')
```

Comment:
I am not sure of that, it should also depends of `move_raw_ids. ....`, `operation_ids` ? 
if when the state is compute but doesn't change, the `_compute_reservation_state` will be call ?

Reply:
I added `move_raw_ids.state` to the depends, but I don't think `operation_ids` should be included. If someone changes the `operation_ids` of a BOM after the MO is confirmed, does it affect the MO? I thought it doesn't since it looks like operation_ids correspond to the work orders and those don't change after MO confirmation. Maybe I'm wrong?

Reply:
You are right, it is the bom operation_ids and the BoM should change after confirmation. IMO, it is correct.